### PR TITLE
Change order of field attribute parsing

### PR DESCRIPTION
**Pull request by do4gr**
always do all scalars first, then all relationfields. The creation of foreign key names during relation attribute parsing depends on the @map attributes on scalar fields already being resolved.

### CODE REVIEW

File: libs/datamodel/core/src/transform/ast_to_dml/db/attributes.rs [hunk 1]
```diff
@@ -51,8 +51,11 @@ pub(super) fn resolve_model_and_field_attributes<'ast>(
         }
     }
 
+    //first go over all scalars since relations depend on them
+    let mut seen_scalars = vec![];
```
Comment (pimeys) [inline libs/datamodel/core/src/transform/ast_to_dml/db/attributes.rs:55]: Why not `HashSet`?

File: libs/datamodel/core/tests/attributes/relations/relations_positive.rs [hunk 2]
```diff
@@ -282,3 +282,71 @@ fn allow_implicit_fk_name_definition() {
         .assert_relation_referenced_fields(&["user_id"])
         .assert_relation_fk_name(Some("Post_user_id_fkey".to_string()));
 }
+
+#[test]
+fn implicit_fk_name_definition_with_mapped_models_and_fields() {
+    let dml = with_named_constraints(
+        r#"
+     model User {
```
Comment (pimeys) [inline libs/datamodel/core/tests/attributes/relations/relations_positive.rs:290]: Indentation would do miracles here. If the parsing goes awry, use `indoc`.

File: libs/datamodel/core/tests/attributes/relations/relations_positive.rs [hunk 3]
```diff
@@ -282,3 +282,71 @@ fn allow_implicit_fk_name_definition() {
...
+        .assert_has_relation_field("user")
+        .assert_relation_referenced_fields(&["user_id"])
+        .assert_relation_fk_name(Some("PostMap_user_id_map_on_post_fkey".to_string()));
+}
+
+#[test]
+fn implicit_fk_name_definition_with_mapped_models_and_fields_other_order() {
+    let dml = with_named_constraints(
+        r#"
+     model User {
```
Comment (pimeys) [inline libs/datamodel/core/tests/attributes/relations/relations_positive.rs:324]: Indentation here too.

**Review by pimeys — COMMENTED** [review]
Quick and dirty, a few fixes and good to go.

**Comment by do4gr** [comment]
Merging now to unblock the release, and doing the style feedback in a separate PR later.

**Comment by tomhoule** [comment]
There was already a hack with `@map` in the first PR to accommodate the default constraint names, I think we should refactor this asap into either lazy default name generation (my preference) or a separate step after attribute parsing.

**Pull request merged**